### PR TITLE
refactor(ivy): minor improvements / cleanup in the DI code

### PR DESCRIPTION
**Pull request by pkozlowski-opensource**

### CODE REVIEW

**Review by mhevery — APPROVED** [review]

**Comment by mhevery** [comment]
[presubmit](http://test/OCL:280215623:BASE:280215671:1573666606311:6997b999)
[ivy presubmit](http://test/OCL:280215623:BASE:280215872:1573666671263:a9335334)

File: packages/core/src/render3/instructions/di.ts [hunk 1]
```diff
@@ -40,15 +41,15 @@ export function ɵɵdirectiveInject<T>(token: Type<T>| InjectionToken<T>): T;
 export function ɵɵdirectiveInject<T>(token: Type<T>| InjectionToken<T>, flags: InjectFlags): T;
 export function ɵɵdirectiveInject<T>(
     token: Type<T>| InjectionToken<T>, flags = InjectFlags.Default): T|null {
-  token = resolveForwardRef(token);
```
Comment (kara) [inline packages/core/src/render3/instructions/di.ts]: Wouldn't this mean that we aren't resolving forward refs for tokens in the fallback case on line 47? Can we write a test to make sure we're not breaking that case?

**Review by kara — CHANGES_REQUESTED** [review]

File: packages/core/src/render3/instructions/di.ts (hunk 1 above)
Comment (pkozlowski-opensource) [reply packages/core/src/render3/instructions/di.ts]: `ɵɵinject` is doing forwardRef resolving so we should be good. But it is always to have a test verifying this! I _was assuming_  that we've got one but I will check if removing forward ref resolution from  `ɵɵinject` breaks any of the existing tests and if not - I will add one.

Comment (kara) [reply packages/core/src/render3/instructions/di.ts]: Ah OK, looks like we are good then!

**Review by kara — APPROVED** [review]
LGTM

**Pull request closed without merging**